charts/argo-rollouts/templates/_helpers.tpl

```diff
@@ -52,6 +52,14 @@ app.kubernetes.io/name: {{ include "argo-rollouts.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
 
+{{/*
+Selector labels: dashboard
+*/}}
+{{- define "argo-rollouts.selectorLabels-dashboard" -}}
+app.kubernetes.io/name: {{ include "argo-rollouts.name" . }}-dashboard
+app.kubernetes.io/instance: {{ .Release.Name }}-dashboard
+{{- end }}
+
```

Comment:
I think we do not need a separate helper for this as we need this only once?

I'd take:
```yaml
spec:
  selector:
    matchLabels:
      {{- include "argo-rollouts.selectorLabels" . | nindent 6 }}
      app.kubernetes.io/component: {{ .Values.dashboard.component }}
```
WDYT?

Reply:
Hi, @mkilchhofer , thanks for your suggestions; dashboard service uses the `argo-rollouts.selectorLabels-dashboard` as well. When I made the following

```yaml
{{- if .Values.dashboard.enabled -}}
apiVersion: v1
kind: Service
metadata:
  name: {{ include "argo-rollouts.fullname" . }}-dashboard
  labels:
    app.kubernetes.io/component: {{ .Values.dashboard.component }}
    {{- include "argo-rollouts.labels" . | nindent 4 }}
  {{- with .Values.serviceAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
spec:
  ports:
  - name: dashboard
    protocol: TCP
    port: 3100
    targetPort: 3100
  selector:
    {{- include "argo-rollouts.selectorLabels" . | nindent 6 }}                         # <--------- Use the shared label
    app.kubernetes.io/component: {{ .Values.dashboard.component }}         # <--------- Additional selector label to distinguish between dashboard and rollout deployment
{{- end }}
```

the above template gives me the error `Error: YAML parse error on argo-rollouts/templates/dashboard/service.yaml: error converting YAML to JSON: yaml: line 21: did not find expected key`. Any idea to fix?

Reply:
Because the selector on the Service is only nindent 4 :-)

Reply:
Thanks! Will send an updated PR.